fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java

```diff
@@ -169,13 +169,19 @@ private double calculateKeySize() {
                 buildMapOp = rightOp;
             }
 
+            ColumnStatistic keyStatistics;
             if (buildMapOp.isColumnRef()) {
-                keySize += rightTableStat.getColumnStatistics().get(buildMapOp).getAverageRowSize();
+                keyStatistics = rightTableStat.getColumnStatistic((ColumnRefOperator) buildMapOp);
             } else {
                 Statistics.Builder allBuilder = Statistics.builder();
                 allBuilder.addColumnStatistics(rightTableStat.getColumnStatistics());
-                ColumnStatistic outputStatistic = ExpressionStatisticCalculator.calculate(buildMapOp, allBuilder.build());
-                keySize += outputStatistic.getAverageRowSize();
+                keyStatistics = ExpressionStatisticCalculator.calculate(buildMapOp, allBuilder.build());
+            }
+
+            if (keyStatistics.isUnknown()) {
+                keySize += keyStatistics.getAverageRowSize();
```

Comment:
Why use averageRowSize when statistics is unknown?

Reply:
error code